src/components/add-event/AddEvent.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {useContext, useState} from "react";
+import {MedicalCalendarContext} from "src/context/calendar-context/MedicalCalendarContext";
+import moment from "moment";
+import DropdownList from "react-widgets/DropdownList";
+import "react-widgets/styles.css";
+import "./add-event.css"
+import {CalendarContextType, CalendarEventType, CalendarEventWatch} from "src/models/Models";
+
+type Props = {
+    selectedDate : Date
+}
+
+const AddEvent = ({selectedDate}: Props) => {
+
+    let types = [
+        new CalendarEventType('Enf', 'Enfermaria'),
+        new CalendarEventType('Amb', 'Ambulatório'),
+        new CalendarEventType('Pla', 'Plantão'),
+        new CalendarEventType('PosP', 'Pós Plantão'),
+        new CalendarEventType('Aula', 'Aula'),
+    ];
+
+
+    const {events, saveEvents} = useContext(MedicalCalendarContext) as CalendarContextType
+    const [type, setType] = useState("")
+    const [watch, setWatch] = useState("")
+    const [watches, setWatches] = useState<CalendarEventWatch[]>([])
+
+    function clearMorningEvents() {
+        return events.filter((current) => {
+                let mCurrDate = moment(current.Date)
+                let mSelectedDate = moment(selectedDate)
+                return !mCurrDate.isSame(mSelectedDate) || current.Type !== 'manha'
+            }
+        )
+    }
+
+    const addEvent = () => {
+        let event = {
+            Date: selectedDate,
+            Type: type,
+            Watch: watch
+        }
+        let eventList
+        if (event.Type === 'PosP') {
+            eventList = clearMorningEvents()
+        } else {
+            eventList = [...events]
+        }
+
+        localStorage.setItem("events", JSON.stringify([...eventList, event]))
+        saveEvents(JSON.parse(localStorage.getItem('events')!))
+    }
+
+    const clearDay = () => {
+        let newContext = events.filter((current) => {
+            let mCurrDate = moment(current.Date)
+            let mSelectedDate = moment(selectedDate)
+
+            return !mCurrDate.isSame(mSelectedDate)
+        })
+        localStorage.setItem("events", JSON.stringify([...newContext]))
+        saveEvents(JSON.parse(localStorage.getItem('events')!))
+    }
+
+    let formattedDate = moment(selectedDate).format('DD/MM/YYYY');
+
+    const filterWatches = (typeId:string) => {
+        let list = new Array<CalendarEventWatch>()
+        if( typeId === 'Enf') {
```

Comment:
Aqui poderia utilizar object literals para deixar o código mais limpo e remover as condicionais. Combinando essa técnica com enum vai deixar o código mais resiliente ainda.

https://dev.to/lukyhenson/substitua-sua-instrucao-switch-e-varios-if-and-else-usando-object-literals-pt-br-4po9